### ISSUE TITLE
Terraform limits cognito verification email body to 140 characters

### ISSUE THREAD

**Issue by bleggett** [terraform-providers/terraform-provider-aws#4050]
### Terraform Version
0.11.5

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_cognito_user_pool

### Terraform Configuration Files
```hcl
resource "aws_cognito_user_pool" "pool"{
  name = "${var.envPrefix}"
  username_attributes = ["email"]
  schema = [
    {
      name = "email"
      attribute_data_type = "String"
      required = true
    },
    {
      name = "reg-code"
      attribute_data_type = "String"
      string_attribute_constraints = {
        min_length = 1
      },
    }
  ]
  tags = {
    Application = "Test"
    AppInstance = "${var.envPrefix}"
  }
  auto_verified_attributes = ["email"]
  verification_message_template = {
    email_subject_by_link = "APP Notification - Account Verification"
    email_message_by_link = "Hello,\n<br><br>\nThanks for signing up!\n<br><br>\nPlease click the link to verify your email address: {##VERIFY EMAIL##}\n<br><br>\nTo know more about APP, please refer to link https://github.com/test/test-core/test\n<br><br>\nBest,<br>\nAPP Team"
    default_email_option = "CONFIRM_WITH_LINK"
  }
  password_policy = {
    minimum_length    = 6
    require_lowercase = true
    require_numbers   = false
    require_symbols   = false
    require_uppercase = false
  }
}

```

### Debug Output
https://gist.github.com/bleggett/d3a79afd6d0d2c2256bb87770b1b965c

### Expected Behavior
Terraform should not limit people to a 140-character email body when AWS has no such limitation.

### Actual Behavior
Terraform fails if I set `verification_message_template.email_message_by_link` to a string longer than 140 characters, because that limit is hardcoded in a `validators.go` function named `validateCognitoUserPoolTemplateEmailMessageByLink`

AWS has no problem with the email body if I use their API directly, and [appears to support up to 20,000 UTF-8 characters](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-email-verification-message-customization.html)

**Comment by bflad** [comment]
The fix for this has been merged into master and will be released with v1.14.0 of the AWS provider, likely by the end of the week.

**Comment by bflad** [comment]
This has been released in [version 1.14.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1140-april-06-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
